Sources/Plot/API/HTMLFormMethod.swift

```diff
@@ -0,0 +1,37 @@
+//
+//  HTMLFormMethod.swift
+//  
+//
+//  Created by Michael Critz on 12/19/19.
+//
+
+import Foundation
+
+/// Describes the form action’s `method`
+public enum HTMLFormMethod: String, CustomStringConvertible {
+    // CORS-safelisted methods
+    /// GET: By REST convention, read a resource.  Requests using GET should only retrieve data.
+    case get
+    /// Like GET, but discared the body. Not a REST convetion.
+    case head
+    /// By REST convention, create a resource.
+    case post
+    
+    // Acceptable methods
+    /// DELETE: by REST convention, delete a resource.
+    case delete
+    /// PUT: by REST convention, update a resource by *replacing* the existing resource with the request body.
+    case put
+    /// OPTIONS: used to describe the communication options for the target resource.
+    case options
+    /// PATCH: by REST convention, update a resource by *modifing* part of the resouces with the request body.
+    case patch
+    
+    // Accodring the the HTML spec for form these forbidden `method`s should never be included: `CONNECT`, `TRACE`, and `TRACK`.
+    
+    public var description: String {
```

Comment:
I would go with `RawRepresentable` (like I proposed in `HTMLFormEnctype `) here as well:

```swift
public enum HTMLFormMethod: String, RawRepresentable {
    // CORS-safelisted methods
    /// GET: By REST convention, read a resource.  Requests using GET should only retrieve data.
    case get = "GET"
    /// Like GET, but discared the body. Not a REST convetion.
    case head = "HEAD"
    /// By REST convention, create a resource.
    case post = "POST"
    
    // Acceptable methods
    /// DELETE: by REST convention, delete a resource.
    case delete = "DELETE"
    /// PUT: by REST convention, update a resource by *replacing* the existing resource with the request body.
    case put = "PUT"
    /// OPTIONS: used to describe the communication options for the target resource.
    case options = "OPTIONS"
    /// PATCH: by REST convention, update a resource by *modifing* part of the resouces with the request body.
    case patch = "PATCH"
}
```

Again I'm feeling like using the `description` property is code smell. Also I have the feeling explicitly defining the enums values (even though they currently are only the key's uppercased representation) would be beneficial, because: if there is a new key which would differ from the value then you're whole implementation might break.

Reply:
Done